solr/core/src/java/org/apache/solr/handler/component/QueryElevationComponent.java

```diff
@@ -376,19 +375,19 @@ protected long getConfigVersion(SolrCore core) {
    *
    * @return The loaded {@link ElevationProvider}; not null.
    */
-  private ElevationProvider loadElevationProvider(SolrCore core) throws Exception {
-    XmlConfigFile cfg;
-    try {
-      cfg = new XmlConfigFile(core.getResourceLoader(), configFileName);
+  private ElevationProvider loadElevationProvider(SolrCore core) throws IOException, SAXException {
+    try (var inputStream = core.getResourceLoader().openResource(configFileName)) {
```

Comment:
I liked the flow previously better -- declare an object holding the parsed input.  And later, not in a try-finally, we call loadElevationProvider.  Not using XmlConfigFile would only mean saying `Document` instead.  A benefit to the previous flow is that the empty file checking would not happen if loadElevationProvider fails, since there's no point.

Reply:
Yea, there's something strange about the flow here, where we open the resource once and then again open the same resource to check if it's empty (line 391)